pkg/sql/sqlbase/validate_test.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sqlbase
+
+import (
+	"reflect"
+	"testing"
+)
+
+type validateStatus int
+
+const (
+	// statusUnvalidated means that a field hasn't been added to any Validate
+	// methods and will fail the test.
+	statusUnvalidated validateStatus = iota
+	// thisFieldReferencesNoObjects means that a field doesn't have any meaningful
+	// references to other objects, and therefore isn't validated.
+	thisFieldReferencesNoObjects
+	// todoIAmKnowinglyAddingTechDebt means that a field wasn't added to
+	// any validation methods, and that you're knowingly adding tech debt! You
+	// must add a justification for this in the map.
+	todoIAmKnowinglyAddingTechDebt
+	// iSolemnlySwearThisFieldIsValidated means that a field was added to a
+	//validate method.
+	iSolemnlySwearThisFieldIsValidated
+)
+
+// validationMap is a structure that contains a "validation status" for every
+// field in all listed descriptors.
+//
+// The purpose of this map is to force people who are adding new descriptor
+// fields to remember to add those fields to the appropriate Validate methods.
+// If you're here because you failed the test at the bottom of this file, you
+// should add an entry for your new field(s) to the map below. Please think
+// carefully when adding your entry, and be truthful - nothing checks these
+// validation statuses but you and your code reviewers.
+//
+// Adding information to Validate is extremely important. Writing corrupted
+// descriptors to disk is a real risk and has real consequences. Please do your
+// part to ensure your features don't have this risk.
+//
+// The validation statuses are descriptively named so that reviewers will take
+// note of what you add to the map :)
+var validationMap = []struct {
+	obj      interface{}
+	fieldMap map[string]validationStatusInfo
+}{
+	{
+		obj: TableDescriptor{},
+		fieldMap: map[string]validationStatusInfo{
+			"Name":             {status: thisFieldReferencesNoObjects},
+			"ID":               {status: thisFieldReferencesNoObjects},
+			"Version":          {status: thisFieldReferencesNoObjects},
+			"ModificationTime": {status: thisFieldReferencesNoObjects},
+			"DrainingNames":    {status: thisFieldReferencesNoObjects},
+			"ParentID":         {status: iSolemnlySwearThisFieldIsValidated},
+			"UnexposedParentSchemaID": {
+				status: todoIAmKnowinglyAddingTechDebt,
+				reason: "initial import: TODO(features)"},
+			"Columns":      {status: iSolemnlySwearThisFieldIsValidated},
```

Comment:
Can you mark this as a TODO? We'll want to look make sure that any user defined types the column points to in fact exist.